airbyte-integrations/connectors/source-google-analytics-data-api/unit_tests/test_source.py

```diff
@@ -97,6 +98,23 @@ def test_check(requests_mock, config_gen, config_values, is_successful, message)
         assert e.value.failure_type == FailureType.config_error
 
 
+def test_missing_metadata(requests_mock):
+    # required for MetadataDescriptor $instance input
+    class TestConfig:
+        config = {
+            "authenticator": None,
+            "property_id": 123,
+        }
+
+    # mocking the url for metadata
+    requests_mock.register_uri("GET", "https://analyticsdata.googleapis.com/v1beta/properties/123/metadata", json={}, status_code=403)
```

Comment:
I'd suggest adding more test cases with a lot of different status code and assert the expected behavior.

Reply:
Added 401 and 403 cases, these are checked by the logic, all other ones are tested on the `check_connection` tests.